File: PokemonApp/app/src/main/java/com/codeprecious/pokemonapp/di/retrofitModule.java
package com.codeprecious.pokemonapp.di;

import com.codeprecious.pokemonapp.retrofit.PokemonApiService;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava3.RxJava3CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

@Module
@InstallIn(SingletonComponent.class)
public class retrofitModule {

    @Singleton
    @Provides
    public static PokemonApiService provideApiService() {
        return new Retrofit.Builder().baseUrl("https://pokeapi.co/api/v2/")
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
                .build().create(PokemonApiService.class);
    }

}

File: FakeDownloadApp/app/src/main/java/com/codeprecious/fakedownloadapp/di/RetrofitModule.kt
package com.codeprecious.fakedownloadapp.di

import com.codeprecious.fakedownloadapp.retrofit.RetrofitApi
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import io.reactivex.rxjava3.plugins.RxJavaPlugins
import retrofit2.Retrofit
import retrofit2.adapter.rxjava3.RxJava3CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
object RetrofitModule {

    @Provides
    @Singleton
    fun provideRetrofitApi(): RetrofitApi {
        return Retrofit.Builder().baseUrl(RetrofitApi.BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
            .build().create(RetrofitApi::class.java)
    }
}